core/src/main/java/com/hyperwallet/android/model/transfermethod/HyperwalletPrepaidCard.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2018 Hyperwallet
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
+ * and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including
+ * without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+ * EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.hyperwallet.android.model.transfermethod;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.hyperwallet.android.util.DateUtil;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * Represents the prepaid card fields
+ */
+
+public class HyperwalletPrepaidCard extends HyperwalletTransferMethod {
+
+    private static final String DEFAULT_CARD_PACKAGE = "DEFAULT";
+
+    /**
+     * Constructor to build prepaid card
+     */
+    public HyperwalletPrepaidCard() {
+        super();
+        setField(TransferMethodFields.TYPE, TransferMethodTypes.PREPAID_CARD);
+        setField(TransferMethodFields.CARD_PACKAGE, DEFAULT_CARD_PACKAGE);
+    }
+
+    public HyperwalletPrepaidCard(@NonNull String cardPackage) {
+        super();
```

Comment:
To avoid duplicate code and other problems, please use "constructor chaining"
```suggestion
    public HyperwalletPrepaidCard(@NonNull String cardPackage) {
        this();
        setField(TransferMethodFields.CARD_PACKAGE, cardPackage);
    }
```
Using that approach can avoid issues, such as in the private constructor, which could end up without a value for the card package field if that field is not part of the map

Reply:
updated